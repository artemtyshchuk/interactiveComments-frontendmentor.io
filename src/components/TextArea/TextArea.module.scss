.input {
  box-sizing: border-box;
  padding: 12px 24px;
  width: 100%;
  height: 100px;
  color: var(--Grayish-Blue, #67727e);
  font-feature-settings: "liga" off, "clig" off;
  border-radius: 8px;
  border: 1px solid var(--Moderate-Blue, #5357b6);

  /* Body (M) */
  font-family: Rubik;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px; /* 150% */
  resize: none;
  outline: none;
  caret-color: var(--Moderate-Blue, #5357b6);

  &:focus {
    border: 1px solid var(--Moderate-Blue, #5357b6);
  }
}

@media (max-width: 576px) {
  .input {
    box-sizing: border-box;
    padding: 12px 12px;
  }
}
