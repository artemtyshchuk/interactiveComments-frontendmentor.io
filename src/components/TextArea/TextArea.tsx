import styles from "./TextArea.module.scss";

interface TextAreaProps {
  value: string;
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  style?: React.CSSProperties;
  placeholder?: string;
}

export const TextArea = ({
  value,
  onChange,
  style,
  placeholder,
}: TextAreaProps) => {
  return (
    <textarea
      placeholder={placeholder}
      className={styles.input}
      value={value}
      onChange={onChange}
      style={style}
    />
  );
};
