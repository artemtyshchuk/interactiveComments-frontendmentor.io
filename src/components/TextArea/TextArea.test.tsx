import { render, screen } from "@testing-library/react";
import { TextArea } from "./TextArea";
describe("TextArea", () => {
  it("should render correctly", () => {
    render(
      <TextArea
        placeholder="test"
        onChange={() => {}}
        value="Test"
        style={{ margin: 0 }}
      />
    );

    expect(screen.getByDisplayValue("Test")).toBeInTheDocument();
    expect(screen.getByDisplayValue("Test")).toHaveStyle({ margin: 0 });
    expect(screen.getByPlaceholderText("test")).toMatchSnapshot();
  });
});
