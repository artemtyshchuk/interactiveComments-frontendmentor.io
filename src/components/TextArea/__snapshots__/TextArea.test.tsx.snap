// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render correctly 1`] = `
<textarea
  class="input"
  data-testid="textArea"
  placeholder="test"
  style="margin: 0px;"
>
  Test
</textarea>
`;
